coriolis/tests/licensing/test_client.py

```diff
@@ -0,0 +1,416 @@
+            },
+            response_key='reservation'
+        )
+
+    def test_add_reservation_invalid_type(self):
+        self.assertRaises(
+            ValueError, self.client.add_reservation,
+            'invalid_reservation_type', mock.sentinel.num_vms
+        )
+
+    @mock.patch.object(licensing_module.LicensingClient, 'add_reservation')
+    def test_add_migrations_reservation(self, mock_add_reservation):
+        result = self.client.add_migrations_reservation(2)
+        self.assertEqual(result, mock_add_reservation.return_value)
+
+        mock_add_reservation.assert_called_once_with(
+            licensing_module.RESERVATION_TYPE_MIGRATION, 2
+        )
+
+    @mock.patch.object(licensing_module.LicensingClient, 'add_reservation')
+    def test_add_replicas_reservation(self, mock_add_reservation):
+        result = self.client.add_replicas_reservation(2)
+        self.assertEqual(result, mock_add_reservation.return_value)
+
+        mock_add_reservation.assert_called_once_with(
+            licensing_module.RESERVATION_TYPE_REPLICA, 2
+        )
+
+    @mock.patch.object(licensing_module.LicensingClient, '_get')
+    def test_get_reservations(self, mock_get):
+        result = self.client.get_reservations()
+        self.assertEqual(result, mock_get.return_value)
+
+        mock_get.assert_called_once_with(
+            '/reservations', response_key='reservations'
+        )
+
+    @mock.patch.object(licensing_module.LicensingClient, '_get')
+    def test_get_reservation(self, mock_get):
+        result = self.client.get_reservation(self.reservation_id)
+        self.assertEqual(result, mock_get.return_value)
+
+        mock_get.assert_called_once_with(
+            f'/reservations/{self.reservation_id}', response_key='reservation'
+        )
+
+    @mock.patch.object(licensing_module.LicensingClient, '_post')
+    def test_check_refresh_reservation(self, mock_post):
+        result = self.client.check_refresh_reservation(self.reservation_id)
+        self.assertEqual(result, mock_post.return_value)
+
+        mock_post.assert_called_once_with(
+            f'/reservations/{self.reservation_id}/refresh',
+            None, response_key='reservation'
+        )
+
+    @mock.patch.object(licensing_module.LicensingClient, '_do_req')
+    def test_delete_reservation(self, mock_do_req):
+        mock_response = self.setup_mock_response(
+            mock_do_req, ok=False, status_code=404)
+
+        with self.assertLogs('coriolis.licensing.client', level=logging.WARN):
+                self.client.delete_reservation(
+                    self.reservation_id, raise_on_404=True)
+
+        mock_do_req.assert_called_once_with(
+            'delete', '/reservations/%s' % self.reservation_id,
+            raw_response=True
+        )
+        mock_response.raise_for_status.assert_called_once()
+
+    @mock.patch.object(licensing_module.LicensingClient, '_do_req')
+    def test_delete_reservation_non_404(self, mock_do_req):
```

Comment:
And, I would simply have this test case named `test_delete_reservation`, since it's represents the normal execution of the method (without any errors).